components/kyma-environment-broker/internal/process/provisioning/inject_btp_operator_credentials_step.go

```diff
@@ -0,0 +1,145 @@
+	return &InjectBTPOperatorCredentialsStep{
+		operationManager:  process.NewOperationManager(os),
+		k8sClientProvider: k8sClientProvider,
+	}
+}
+
+func (s *InjectBTPOperatorCredentialsStep) Name() string {
+	return "Inject_BTP_Operator_Credentials"
+}
+
+func (s *InjectBTPOperatorCredentialsStep) Run(operation internal.Operation, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+
+	if operation.RuntimeID == "" {
+		log.Error("Runtime ID is empty")
+		return s.operationManager.OperationFailed(operation, "Runtime ID is empty", nil, log)
+	}
+
+	clusterID := operation.InstanceDetails.ServiceManagerClusterID
+	if clusterID == "" {
+		clusterID = uuid.NewString()
+		updatedOperation, backoff, _ := s.operationManager.UpdateOperation(operation, func(op *internal.Operation) {
+			op.InstanceDetails.ServiceManagerClusterID = clusterID
+		}, log)
+		if backoff != 0 {
+			log.Error("cannot save cluster ID")
+			return updatedOperation, backoff, nil
+		}
+	}
+
+	if operation.ProvisioningParameters.ErsContext.SMOperatorCredentials == nil {
+		log.Error("no credentials set")
+		return s.operationManager.OperationFailed(operation, "no credentials set", nil, log)
+	}
+
+	secret := s.prepareSecret(operation.ProvisioningParameters.ErsContext.SMOperatorCredentials, clusterID)
+
+	if operation.K8sClient == nil {
+		log.Error("kubernetes client not set")
+		return s.operationManager.OperationFailed(operation, "kubernetes client not set", nil, log)
+	}
+
+	if err := s.createOrUpdateSecret(operation.K8sClient, secret, log); err != nil {
+		err = kebError.AsTemporaryError(err, "failed create/update of the secret")
+		return operation, updateSecretBackoff, nil
+	}
+	return operation, 0, nil
+}
+
+func (s *InjectBTPOperatorCredentialsStep) prepareSecret(credentials *internal.ServiceManagerOperatorCredentials, clusterID string) *apicorev1.Secret {
+	return &apicorev1.Secret{
+		TypeMeta: v1.TypeMeta{Kind: "Secret"},
+		ObjectMeta: v1.ObjectMeta{
+			Name:        secretName,
+			Namespace:   secretNamespace,
+			Labels:      labels,
+			Annotations: annotations,
+		},
+		StringData: map[string]string{
+			"clientid":       credentials.ClientID,
+			"clientsecret":   credentials.ClientSecret,
+			"sm_url":         credentials.ServiceManagerURL,
+			"tokenurl":       credentials.URL,
+			"tokenurlsuffix": credentials.XSAppName,
+			"cluster_id":     clusterID},
+		Type: apicorev1.SecretTypeOpaque,
+	}
+}
+
+func (s *InjectBTPOperatorCredentialsStep) createOrUpdateSecret(k8sClient client.Client, parametersBasedSecret *apicorev1.Secret, log logrus.FieldLogger) error {
+
+	clusterSecret := apicorev1.Secret{}
+	err := k8sClient.Get(context.Background(), client.ObjectKey{Namespace: secretNamespace, Name: secretName}, &clusterSecret)
+	if err != nil {
+		return s.createOrRetry(k8sClient, parametersBasedSecret, err, log)
+	}
+	if isNotGeneratedByKEB(clusterSecret) {
+		log.Warn("the secret was not created by KEB and its data will be overwritten")
```

Comment:
nit: consider logging the `secretName` to make reading the logs a bit more obvious

Reply:
Added.